From ee5ce3f0637d5533e6e1f68e6a7e7f846ff742c4 Mon Sep 17 00:00:00 2001
From: Bryce Harrington <bryce@canonical.com>
Date: Thu, 2 Jul 2020 01:50:48 +0000
Subject: [PATCH] Treat devtmpfs and squashfs as dummy filesystems

Avoid displaying snaps in output from df and other tools, by excluding
squashfs filesystems.  Exclude devtmpfs filesystems in output from df
and other tools since it is a dummy filesystem.

[juliank: Update to add fuse.portal]

Origin: upstream, https://debbugs.gnu.org/cgi/bugreport.cgi?bug=37702#68
Bug: https://debbugs.gnu.org/cgi/bugreport.cgi?bug=37702
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/apt/+bug/1756595
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/coreutils/+bug/1219529
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/coreutils/+bug/1905623

---
 lib/mountlist.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/lib/mountlist.c
+++ b/lib/mountlist.c
@@ -164,7 +164,10 @@
 
 #define ME_DUMMY_0(Fs_name, Fs_type)            \
   (strcmp (Fs_type, "autofs") == 0              \
+   || strcmp (Fs_type, "devtmpfs") == 0         \
+   || strcmp (Fs_type, "fuse.portal") == 0      \
    || strcmp (Fs_type, "proc") == 0             \
+   || strcmp (Fs_type, "squashfs") == 0         \
    || strcmp (Fs_type, "subfs") == 0            \
    /* for Linux 2.6/3.x */                      \
    || strcmp (Fs_type, "debugfs") == 0          \
